src/main/java/edu/ucla/library/iiif/fester/CsvParser.java

```diff
@@ -304,6 +328,65 @@ private void extractPageMetadata(final String... aRow) throws CsvParsingExceptio
         return aRow;
     }
 
+    /**
+     * Checks for A/V metadata in rows that represent v2 canvases, and returns the row if none is found.
+     *
+     * @param aRow A row from the metadata CSV
+     * @param aPath A path to a CSV file
+     * @return The row
+     * @throws CsvParsingException If the row represents a v2 canvas and contains any A/V metadata
+     * @throws MimeTypeParseException
+     */
+    @SuppressWarnings("unchecked")
+    private String[] checkIiifPresApiCompatibility(final String[] aRow, final Path aPath, final String aIiifVersion)
+            throws CsvParsingException {
+        final String rowId = getMetadata(aRow, myCsvHeaders.getItemArkIndex()).get();
+
+        final Optional<Integer> mediaWidth =
```

Comment:
Heh, was just re-debating the internal use of Optional(s) again with myself with the ticket I'm working on. They do make using things that are... well, optional... easier.